### ISSUE TITLE
upsertMany insertMany

### ISSUE THREAD

**Issue by dufoli** [jsforce/jsforce#1194]
Currently insert and upsert do a single call for each records.
updateMany function permit to use composite api to update massively some data but there are no equivalent for upsert.

insertMany : composite API POST
https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_create.htm

upsertMany : composite API PATCH
https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_upsert.htm

**Comment by dufoli** [comment]
and delete many ;-)

**Comment by paustint** [comment]
As mentioned [in the docs](https://jsforce.github.io/document/#operation-for-multiple-records), as long as you are using JSForce 1.9+ and Salesforce API 41.0+ it will call the Collections API if you pass in more than one record.

If you are not explicitly setting the API version when you initialize JSForce, then you will probably be using an older API version.

In addition, you can make an API request you want and call these APIs yourself if you need/want more control as mentioned here: https://github.com/jsforce/jsforce/issues/934#issuecomment-546714813

**Comment by dufoli** [comment]
@paustint yes collection API support insert, update and delete but not upsert. I will do a pull request.

**Comment by paustint** [comment]
@dufoli - You may not want to spend time on a PR because it probably will never be merged and would be a waste of your time.

**Comment by paustint** [comment]
You can also use the generic request api to handle this use-case: #934 (comment)

**Comment by dufoli** [comment]
Sorry already done. too fast ;-p
